src/main/java/net/imglib2/img/basictypeaccess/ArrayDataAccessFactory.java

```diff
@@ -0,0 +1,128 @@
+import net.imglib2.type.NativeType;
+import net.imglib2.type.PrimitiveType;
+import net.imglib2.type.PrimitiveTypeInfo;
+
+/**
+ * Given a {@link PrimitiveType} and {@link AccessFlags} creates a specific
+ * {@link ArrayDataAccess}. For example, {@code BYTE} with flags {@code DIRTY}
+ * and {@code VOLATILE} specifies {@link DirtyVolatileByteArray}.
+ *
+ * @author Tobias Pietzsch
+ */
+public class ArrayDataAccessFactory
+{
+	public static < T extends NativeType< T >, A > A get(
+			final T type,
+			final AccessFlags... flags )
+	{
+		return get( type.getPrimitiveTypeInfo().getPrimitiveType(), flags );
+	}
+
+	public static < A extends ArrayDataAccess< A > > A get(
+			final PrimitiveTypeInfo< ?, ? super A > primitiveTypeInfo,
+			final AccessFlags... flags )
+	{
+		return get( primitiveTypeInfo.getPrimitiveType(), flags );
+	}
+
+	@SuppressWarnings( "unchecked" )
+	public static < A extends ArrayDataAccess< A > > A get(
+			final PrimitiveType primitiveType,
+			final AccessFlags... flags )
+	{
+		final boolean dirty = AccessFlags.isDirty( flags );
+		final boolean volatil = AccessFlags.isVolatile( flags );
+		switch ( primitiveType )
+		{
+		case BYTE:
+			return dirty
+					? ( volatil
+							? ( A ) new DirtyVolatileByteArray( 0, true )
+							: ( A ) new DirtyByteArray( 0 ) )
+					: ( volatil
+							? ( A ) new VolatileByteArray( 0, true )
+							: ( A ) new ByteArray( 0 ) );
+		case CHAR:
+			return dirty
+					? ( volatil
+							? ( A ) new DirtyVolatileCharArray( 0, true )
+							: ( A ) new DirtyCharArray( 0 ) )
+					: ( volatil
+							? ( A ) new VolatileCharArray( 0, true )
+							: ( A ) new CharArray( 0 ) );
+		case DOUBLE:
+			return dirty
+					? ( volatil
+							? ( A ) new DirtyVolatileDoubleArray( 0, true )
+							: ( A ) new DirtyDoubleArray( 0 ) )
+					: ( volatil
+							? ( A ) new VolatileDoubleArray( 0, true )
+							: ( A ) new DoubleArray( 0 ) );
+		case FLOAT:
+			return dirty
+					? ( volatil
+							? ( A ) new DirtyVolatileFloatArray( 0, true )
+							: ( A ) new DirtyFloatArray( 0 ) )
+					: ( volatil
+							? ( A ) new VolatileFloatArray( 0, true )
+							: ( A ) new FloatArray( 0 ) );
+		case INT:
+			return dirty
+					? ( volatil
+							? ( A ) new DirtyVolatileIntArray( 0, true )
+							: ( A ) new DirtyIntArray( 0 ) )
+					: ( volatil
+							? ( A ) new VolatileIntArray( 0, true )
+							: ( A ) new IntArray( 0 ) );
+		case LONG:
+			return dirty
+					? ( volatil
+							? ( A ) new DirtyVolatileLongArray( 0, true )
+							: ( A ) new DirtyLongArray( 0 ) )
+					: ( volatil
+							? ( A ) new VolatileLongArray( 0, true )
+							: ( A ) new LongArray( 0 ) );
+		case SHORT:
+			return dirty
+					? ( volatil
+							? ( A ) new DirtyVolatileShortArray( 0, true )
+							: ( A ) new DirtyShortArray( 0 ) )
+					: ( volatil
+							? ( A ) new VolatileShortArray( 0, true )
+							: ( A ) new ShortArray( 0 ) );
+		default:
+			throw new IllegalArgumentException();
+		}
+	}
```

Comment:
This part is not nice, and hopefully the is a way to replace it.
ArrayDataAccessFactory is a "list" of all ArrayAccess classes.
Using such list will make it harder to add ArrayAccesses in the future.
I will think about a solution tomorrow